src/core_ocean/shared/mpas_ocn_gm.F

```diff
@@ -342,177 +364,175 @@ subroutine ocn_gm_compute_Bolus_velocity(diagnosticsPool, meshPool, scratchPool)
+            end do
          end do
-         !$omp end do
-      end do  ! iter
 
-      nCells = nCellsArray ( 2 )
-      ! first, compute tapering across full domain based on a maximum allowable slope
-      !$omp do schedule(runtime) private(k)
-      do iCell=1,nCells
-        do k = 1, maxLevelCell(iCell)
-          relativeSlopeTaperingCell(k,iCell) = min(1.0_RKIND, config_max_relative_slope / (relativeSlopeTopOfCell(k,iCell)+epsGM))
-        end do
-      end do
-      !$omp end do
-
-      ! now further taper in the boundary layer
-      ! vertical (k33) tapering starts at 2*OBL, increases linearly to OBL and is held uniform across OBL
-      ! rtmp = 1 @ zMid = -2.0*OBL, rtmp = 0 @ zMid = -OBL
-      if(config_use_Redi_surface_layer_tapering) then
-         nCells = nCellsArray ( 2 )
-         !$omp do schedule(runtime) private(k, rtmp)
-         do iCell=1,nCells
-           do k = 1, maxLevelCell(iCell)
-             rtmp = -zMid(k,iCell)/max(config_Redi_surface_layer_tapering_extent,boundaryLayerDepth(iCell)+epsGM)
-             rtmp = max(0.0_RKIND,rtmp)
-             rtmp = min(1.0_RKIND,rtmp)
-             relativeSlopeTaperingCell(k,iCell) = rtmp*relativeSlopeTaperingCell(k,iCell)
-           end do
-         end do
-         !$omp end do
-      endif ! config_use_Redi_surface_layer_tapering
-
-      ! now further taper in the boundary layer
-      ! vertical (k33) tapering starts at 2*OBL, increases linearly to OBL and is held uniform across OBL
-      ! rtmp = 1 @ zMid = zMid(maxLevelCell) + config_Redi_bottom_layer_tapering_depth, rtmp = 0 @ zMid = zMid(maxLevelCell)
-      if(config_use_Redi_bottom_layer_tapering) then
-         nCells = nCellsArray ( 2 )
-         !$omp do schedule(runtime) private(k, rtmp)
-         do iCell=1,nCells
-           do k = 1, maxLevelCell(iCell)
-             rtmp = (zMid(k,iCell)-zMid(maxLevelCell(iCell),iCell))/(config_Redi_bottom_layer_tapering_depth+epsGM)
-             rtmp = max(0.0_RKIND,rtmp)
-             rtmp = min(1.0_RKIND,rtmp)
-             relativeSlopeTaperingCell(k,iCell) = rtmp*relativeSlopeTaperingCell(k,iCell)
-           end do
+         ! Normalize k33
+         do k = 2, maxLevelCell(iCell)
+            k33(k,iCell) = k33(k,iCell) / k33Norm(k)
+                !print '(a,2i5,1000e9.2)', 'iCell,k,Up',iCell,k,slopeTriadUp(k,:,iCell)
+                !print '(a,2i5,1000e9.2)', 'iCell,k,Dn',iCell,k,slopeTriadDown(k,:,iCell)
+if (k33Norm(k)<1e-15) then
+    print *, 'WARNING: k33Norm(k)<1e-15'
+                !print '(a,100i9)', 'k,iCell: ',k,iCell
+                !print '(a,100e9.2)', 'k33: ',k33(k,iCell)
+                !print '(a,100e9.2)', 'k33Norm: ',k33Norm(k)
+                !print '(a,100f5.2)', 'slopeTaper: ',slopeTaper(k)
+                !print '(a,1000e9.2)', 'slopeTriadUp: ',slopeTriadUp(k,:,iCell)
+                !print '(a,1000e9.2)', 'slopeTriadDown: ',slopeTriadDown(k,:,iCell)
+endif
```

Comment:
maybe remove this debugging print statements?

Reply:
Removed, thanks.